app/uk/gov/hmrc/securemessage/controllers/models/generic/ApiConversation.scala

```diff
@@ -47,34 +47,29 @@ object ApiConversation {
       messages = conversation.messages.map(message => convertToApiMessage(conversation, message, identifier))
     )
 
+  private def isSender(sender: Option[Participant], reader: Option[Participant]) =
```

Comment:
can't you use cats: sender === reader ?

Reply:
we want to compare just Id . Will it work ?